### ISSUE TITLE
Slow down 1st layer above bridge

### ISSUE THREAD

**Issue by AndreySemjonov** [SoftFever/OrcaSlicer#3847]
Would like to have option to slow down 1st layer above bridge to make better contact between bridge and next layer.
If I have fast printing speed (200mm/s+) than I think this affect next layer quality and how bridge attaches to next layer.
Same function as Cura have - Bridge (Second/Third) skin speed (density, flow).

**Comment by Tianziegler** [comment]
i would also be happy to have such a feature when printing a bridge it looks fine but as soon as the second layer gets printed the bridge starts to sag by a lot. Printing rPet at 260°C with a 0.5mm nozzle. Bridge speed 15mm/s and internal solid infill at 400mm/s. Would be good to be able to print two layer bridges for long distances or reduce the speed for the first layer ontop of the bridge and also be able to set a fan speed for that layer.

**Comment by aranadam** [comment]
Please add this feature!!!

**Comment by Ataraxia-Mechanica** [comment]
As an owner of a 500mm/s printer, please add this alongside slowing down for internal solid infill overhangs: #5003

**Comment by Rodamyot** [comment]
I'd love to have this to. Sure I can do multiple layer seting and change speed but the goal in a slicer is to be able to identify trouble in printing parts and solve it. the surface over bridge is so so and speed printing or even normal speed printing need a different parameter to compensate for infill over bridge sirface, like more flow extrusion and slowing speed.

**Comment by wbagnall** [comment]
I second this. An above-bridge speed setting is crucial for bridge quality. OrcaSlicer is unusable for bridging without this feature.

**Comment by GabrieleCzz** [comment]
I also would love this feature!
I made some tests of bridging and one of them was comparing slow speed on top of the bridge w.r.t. normal speed (in both cases the bridge speed was slow). The difference was pretty noticeable!
So, to have good bridges without reducing overall speed (or make manual modifications) it's really necessary to have this feature at least on the first layer above the bridge (maybe also on the second one).

**Comment by leokeba** [comment]
Yes please, this feature is very much needed, especially on fast printers when using low layer heights. My Bambu Lab P1S makes nice bridges but they get destroyed as soon as the next layer prints on top of them at full speed.

**Comment by Bischi1997** [comment]
I would love to see this feature as well :)

**Comment by stanpete78** [comment]
Please do this. On fast printers the bridges are so much worse thn the need to be.

**Comment by boromyr** [comment]
+1

**Comment by cwiegert** [comment]
Yes Please, such a powerful feature for using today's high speed filaments

**Comment by oleksii-suprun** [comment]
I have found a workaround and used the "Height Range Modifier" feature to slow down the infill over the bridge, which is inconvenient. Having a configuration option would be better.

**Comment by jamincollins** [comment]
Just another user requesting this same feature.
:+1:

**Comment by ph0tic** [comment]
yeah. me too pls.

**Comment by ph0tic** [comment]
I do try to print as fast as possible, and i find when the external layer comes back across the bridge, it knocks off the edge on occasion. that really, really sucks.

**Comment by igor-rubis** [comment]
+1 it would be great to have this feature

**Comment by xateesix** [comment]
Not just great to have it, absolutely necessary.

**Comment by ebrune81** [comment]
I've been struggling with this exact problem. Having to slow down the entire infill speed for just the layer above bridge layer seems crazy.

**Comment by sutty5** [comment]
This feature is a must. Multi-layer bridge settings are needed. This is because, for those printing at high speeds (which is becoming more common), when a bridge layer is printed, the next layer on top of the bridge is shown as "internal solid infill". For which, I have mine set at high speed, which, is usually fine. However, for medium to long distance bridges, printing over the top of a bridge layer at 400mm/s and max flow rate completely screws the bridge layer. There needs to be a multi-layer bridge setting. Such that one can ensure the "solid infill" layer on top of a freshly laid bridge is printed slower, ensuring the bridge is properly set and reinforced before proceeded to speed over it with a high speed solid infill layer. I think it should go like so... Bridge Initial Layer, Bridge Final Layer, Internal Solid Infill.

**Comment by Bromim** [comment]
I agree, adjusting the print speed of the layer on the bridge is necessary.

**Comment by LamarcLS** [comment]
+1

**Comment by igiannakas** [comment]
Hi all

Some ground work has been done on this PR here: https://github.com/SoftFever/OrcaSlicer/pull/8263

There is of course plenty more to be done, but the "hard work" of identifying and segmenting the regions above bridges is now complete.

**Comment by igiannakas** [comment]
**Test results from that PR:**
Left - Old / Right - Two bridge layers enabled
Bridge is significantly better
![IMG_5467](https://github.com/user-attachments/assets/a11a88e1-6e33-460c-af34-40825f2a3218)

**Comment by stefansjs** [comment]
Can I request a change to the implementation? It seems like it's excluding long narrow bridges, where slowing down is more critical for me.

![Image](https://github.com/user-attachments/assets/e0784b7f-0de1-4396-a53b-71d4a43c0fc1)
![Image](https://github.com/user-attachments/assets/db682ff1-1c7e-491e-a3ab-edcecb476a20)

**Comment by igiannakas** [comment]
@stefansjs You’re using classic and the bridge you’re seeing is not a bridge, but gap fill. Either make the extrusion width narrower to actually fit a bridge or reduce number of walls.

This bridge area is too thin hence gets filtered out by the algorithm too.

Don’t forget a bridge is coloured blue and it needs to be at least 1-2 lines wide for it to “count”.

**Comment by igiannakas** [comment]
Can you post a screenshot of the layer below the second bridge layer? Ie the first bridge layer please?

**Comment by stefansjs** [comment]
Here's the two adjacent layers. This happens for each bridging leg in the model.

![Image](https://github.com/user-attachments/assets/715b11cf-6e6c-4fbd-8e53-357f60c023a2)
![Image](https://github.com/user-attachments/assets/d45af1ea-02fc-4eef-8349-54d3f8778e9b)

My understanding from the tooltip explanation is that Orca Slicer refuses to generate an *extra* layer if the bridge is too narrow. "Please note that small bridges that are shorter or narrower than the set number of perimeters will be skipped as they would not benefit from a second bridge layer." It seems like this is intended functionality. I'm just hoping for a more permissive extra layer.

**Comment by igiannakas** [comment]
Can you post the project file please?

**Comment by stefansjs** [comment]
[bridging_test_50mm.3mf.gz](https://github.com/user-attachments/files/18982975/bridging_test_50mm.3mf.gz)

Should I open a new issue?

**Comment by igiannakas** [comment]
Let me investigate first and will let you know

**Comment by datspike** [comment]
Same issue as last two comments. Fast walls on the layer next to the first bridge layer ruin the bridge underneath

**Comment by amccoig** [comment]
...Same issue as last 3 comments. The PR is almost there but the walls on the additional bridge layers either need to be removed or at least slowed to match bridge speed and cooling settings.

1st bridge
<img width="645" height="754" alt="Image" src="https://github.com/user-attachments/assets/ac1eae56-d037-4ee6-914c-cf2169198909" />

2nd bridge
<img width="629" height="740" alt="Image" src="https://github.com/user-attachments/assets/1d05ceac-e4ab-4a60-97d2-bcfac50c482b" />

As of now its near impossible to print anything in the PETG I'm using if it has any bridging beyond tiny micro overhang type bridges. 

So far Orca has become my favorite slicer but this is killing it for me.

**Comment by SilencedFrost** [comment]
same problem, my first bridge is looking really nice now, but on the second layer, the fast wall ruins it, having a slow down option for first layer above bridge would be wonderful

1st bridge layer

<img width="1123" height="894" alt="Image" src="https://github.com/user-attachments/assets/24138078-7f6d-4ae7-81c7-536ef08e2362" />

2nd bridge layer

<img width="1131" height="889" alt="Image" src="https://github.com/user-attachments/assets/a18cd2cf-dbc3-485d-a657-1c988254fc7e" />

**Comment by stefansjs** [comment]
Unfortunately this ticket is done and merged. The developer hasn't been back to make changes, and now that it's released I didn't expect them to. I think somebody is going to have to open a new pr with the fix. I would love to say that it was me except I still haven't gotten orcaslicer to compile and at this point I'm probably not getting over that hump any time soon.

Hopefully future developers are more willing to test, or at least parameterize assumptions like this one. It would have been nice to have some way to deal with this limitation besides opening a PR.

**Comment by igiannakas** [comment]
If you’re referring to the walls, on the bridge layer these are classed as overhang walls. Not part of the bridge in the code. Hence they use the bridge / overhang speed. 

The second layer considers them as walls, again not part of the bridge.

Changing this is a fundamental re-write as it impacts the definition of a bridge (the light blue bit in the slicer preview vs the walls)

**Comment by igiannakas** [comment]
Basically the scope of the bridging code only looks at the bridge, not the walls. Changing that is a rewrite on what is classified as a wall. As the walls are extruded parallel to the walls underneath quality due to increased speeds should not suffer (as also seen in my PR image test). If it does, cooling improvements need to be explored.

Code wise the only place that changes wall speeds above overhangs is the curled perimeter detection. That is were logic like this could be potentially implemented if anyone fancies exploring this.

**Comment by stefansjs** [comment]
I'm referring to this line in code where you filter out bridge lines that you think aren't going to matter 

https://github.com/SoftFever/OrcaSlicer/pull/8263/files#diff-3ec83cd4f72e61c92f69178305a0298e90e2d8b43a3d88ca2f8edccde4e02c7aR1487

PrintObject.cpp:1492 has comments explaining why you filter out short segments a.
PrintObject.cpp:1502 filters by internal line type. 

I'd like to think this is trivial to test, but like I said I haven't gotten the slicer to build on my machine and I haven't had the time to figure out why. Did you test different filter operations? Your code explains your conclusions but doesn't explaining your testing procedure. Obviously you don't want to clutter the code with comments like that, but I couldn't figure out if there were other documents that explain your rationale

**Comment by igiannakas** [comment]
The surface area of the bridge does dictate whether you need a second bridge layer. This is used to filter out trivial bridges as they don’t tend to sag or break. 

I’ve tested this with Voron parts where the created bridges are small - for example the flat top of a screw hole. There adding a second bridge layer doesn’t do anything to improve quality as the bridge didn’t separate at all (as it’s too small.). If you have a specific part that is giving you trouble post a pic and I can post a PR that removes that constraint for you to test if it is beneficial to reduce the filtered area.

**Comment by SilencedFrost** [comment]
> Unfortunately this ticket is done and merged. The developer hasn't been back to make changes, and now that it's released I didn't expect them to. I think somebody is going to have to open a new pr with the fix. I would love to say that it was me except I still haven't gotten orcaslicer to compile and at this point I'm probably not getting over that hump any time soon.
> 
> Hopefully future developers are more willing to test, or at least parameterize assumptions like this one. It would have been nice to have some way to deal with this limitation besides opening a PR.

I see, I would like to contribute too, I looked through the source code, and cpp is not within my skillset, I'll contribute one day if possible

**Comment by derei** [comment]
I agree that we need explicit control for the first n solid infill layers applied on top of a bridge.
I believe this must be made of actuality because even "extra bridge layers" is not a good fix. And it mostly happens on short bridges like cut-outs, holes, where a fast printer will be too aggressive. And slowing down the solid infill globally is not a good way to do it - it basically neuters the "fast printer" advantage.

This could be similar to how "slow layers" is implemented: select  a speed and flow ratio override for the 1st solid infill applied exclusively on the bridge, and then a layers count until the speed gets back to nominal.

Example:
80% speed over 2 layers would mean:
1st infill on top of the bridge: 80% speed
2nd infill: 90% speed
3rd infill: back to global settings
